CMakeLists.txt

```diff
@@ -167,3 +167,24 @@ function(mbed_generate_bin_hex target)
         VERBATIM
     )
 endfunction()
+
+#
+# Configure Mbed OS CMake application (default).
+#
+function(mbed_configure_application target)
```

Comment:
We already have `mbed_configure_app_target()`. See line 100 of this file. maybe just add more function calls to it. You wont need to modify the examples because they already call that function.

Reply:
I want to unify component/app functionality, so both should be named similar. 

```
mbed_configure_application/mbed_configure_component
```

I can rename `mbed_configure_app_target` and update what it provides, I'll check.

Reply:
I removed  the old one (I can revert) , asking for "unification" of these 2